File: tests/js/array_iteration_methods/reduce_sum.js
// vybe-test: js/array_iteration_methods/reduce_sum
// origin: languages/js/tests/js/test_array_iteration_methods.rs

function __line(...args) {
    // console.log joins its arguments with a single space. String() is the
    // coercion Vybe's logging host applies to each one.
    return args.map(String).join(" ");
}

function __check(got, want) {
    if (got !== want) {
        console.log("FAIL: want [" + want + "] got [" + got + "]");
        throw new Error("assertion failed");
    }
}

const sum = [1, 2, 3, 4, 5].reduce((acc, x) => acc + x, 0);
__check(__line(sum), "15");

File: tests/fortran/reduce_intrinsic/reduce_sum.f90
! vybe-test: fortran/reduce_intrinsic/reduce_sum
! origin: languages/fortran/tests/fortran/test_fortran2018.rs

program test
    integer :: a(5) = [1, 2, 3, 4, 5]
    integer :: total
    total = reduce(a, operator(+))
    print *, total
end program test
